cmd/bfgd/bfgd.go

```diff
@@ -148,61 +153,100 @@ func HandleSignals(ctx context.Context, cancel context.CancelFunc, callback func
 
 	select {
 	case <-ctx.Done():
-	case s := <-signalChan: // First signal, cancel context.
+	case s := <-signalChan:
 		if callback != nil {
-			callback(s) // Do whatever caller wants first.
-			cancel()
+			callback(s)
 		}
+		cancel()
 	}
-	<-signalChan // Second signal, hard exit.
+	<-signalChan
+	log.Errorf("Received second termination signal, forcing exit.")
 	os.Exit(2)
 }
 
-func _main() error {
-	// Parse configuration from environment
-	if err := config.Parse(cm); err != nil {
-		return err
+// initializeLogger configures logging for the application.
+func initializeLogger(logLevel string) error {
```

Comment:
I don't think this function is necessarily needed. Could you inline the `loggo.ConfigureLoggers` call instead? 🤔